pkg/daemon/sync.go

```diff
@@ -20,13 +21,15 @@ import (
 	"github.com/fluxcd/flux/pkg/update"
 )
 
-// revisionRatchet is for keeping track of transitions between
+// ratchet is for keeping track of transitions between
 // revisions. This is slightly more complicated than just setting the
 // state, since we want to notice unexpected transitions (e.g., when
 // the apparent current state is not what we'd recorded).
-type revisionRatchet interface {
-	Current(ctx context.Context) (string, error)
-	Update(ctx context.Context, oldRev, newRev string) (bool, error)
+type ratchet interface {
+	CurrentRevision(ctx context.Context) (string, error)
+	CurrentResources(ctx context.Context) (map[string]resource.Resource, error)
```

Comment:
Can you think of any circumstances in which this will need a context?
(The current revision needs one because it does git operations)

Reply:
Actually, not.
I wrote this function signature just for alignment with the signature of `CurrentRevision()`.

I have just fixed the signature not to require context.
Thank you for your comment!